src/Core__Array.res

```diff
@@ -194,7 +216,22 @@ let filterMapU = (a, f) => {
 
 let filterMap = (a, f) => filterMapU(a, (. a) => f(a))
 
-// TODO: Change this implementation?
-let flatMap = (a, f) => []->concatMany(map(a, f))
+let keepSome = filterMap(_, x => x)
+
+@send external flatMap: (array<'a>, 'a => array<'b>) => array<'b> = "flatMap"
+
+let findMap = (arr, f) => {
```

Comment:
There was a discussion in #4 where we decided not to add this function for now.

Reply:
I'm open to revisiting that decision though. I still think `findMap` is useful and along the lines of the type of "extra" functionality we could support in `Core`, so if more agree then I don't see any harm in adding it.

Reply:
Not sure what the threshold is, but I've mentioned two use cases here that weren't mentioned in that issue at least. I think this makes more sense in rescript than in JS because we're dealing explicitly with `option`s, whereas making `null` values significant in JS doesn't work since they don't compose (nest) properly.

Reply:
Yeah I think adding it won't hurt, given that it's about handling options, and we're trying to make that as smooth as we can. Ok with you @cknitt ?

Reply:
👍